cmd/network/verification.go

```diff
@@ -429,6 +439,101 @@ func (e *EgressVerification) getSubnetIds(ctx context.Context) ([]string, error)
 	// For non-PrivateLink BYOVPC clusters, provided subnets are 50/50 public/private subnets, so make the user decide for now
 	// TODO: Figure out via IGW/NAT GW/Route Tables
 	return nil, fmt.Errorf("unable to determine which non-PrivateLink BYOVPC subnets are private yet, please check manually and provide the --subnet-id flag")
+
+}
+
+// This function checks the gateway attached to the subnet and returns true if the subnet starts with igw- (for InternetGateway) and has a route to 0.0.0.0/0
+func (e *EgressVerification) isSubnetPublic(ctx context.Context, subnetID string) (bool, error) {
```

Comment:
Some of the logic here looks exactly the same as the logic in https://github.com/fahlmant/osdctl/blob/master/cmd/cluster/cpd.go#L125

Is it possible to pull out the shared logic into a single function?